owncloudApp/src/main/java/com/owncloud/android/providers/WorkManagerProvider.kt

```diff
@@ -82,6 +85,22 @@ class WorkManagerProvider(
             .enqueueUniquePeriodicWork(AVAILABLE_OFFLINE_PERIODIC_WORKER, ExistingPeriodicWorkPolicy.KEEP, availableOfflinePeriodicWorker)
     }
 
+    fun enqueueAccountDiscovery(accountName: String) {
+        val constraintsRequired = Constraints.Builder().setRequiredNetworkType(NetworkType.CONNECTED).build()
+
+        val inputData = workDataOf(
+            AccountDiscoveryWorker.KEY_PARAM_DISCOVERY_ACCOUNT to accountName,
+        )
+
+        val accountDiscoveryWorker = OneTimeWorkRequestBuilder<AccountDiscoveryWorker>()
+            .setInputData(inputData)
+            .addTag(accountName)
```

Comment:
Since `accountName` is added as a tag, this worker will be cancelled as well in `CancelTransfersFromAccountUseCase` together with all the account-related transfers 👍